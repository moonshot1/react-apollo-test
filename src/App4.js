import React, { Component } from 'react'

export default class App4 extends Component {

  constructor(props){
    super(props);
    this.dummyMethod = this.dummyMethod.bind();
  }

  dummyMethod(){
    console.log("it works");
  }

  render() {
    return (
      <div>
        <h1>Hello World</h1>
        <button onClick={this.dummyMethod}>Call Method</button>
      </div>
    )
  }
}
