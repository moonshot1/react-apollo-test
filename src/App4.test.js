import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import App4 from './App4';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('App4', () => {
  it('should render correctly', () => {
    const output = shallow(
      <App4 />
    );

    console.log(output.instance());
    //console.log(output.instance().dummyMethod);

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});