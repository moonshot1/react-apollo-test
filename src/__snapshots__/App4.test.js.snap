// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App4 should render correctly 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <button
    onClick={[Function]}
  >
    Call Method
  </button>
</div>
`;
